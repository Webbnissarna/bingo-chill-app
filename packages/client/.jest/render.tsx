import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
