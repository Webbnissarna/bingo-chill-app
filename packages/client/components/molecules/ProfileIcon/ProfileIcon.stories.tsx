import type { Story, StoryDefault } from "@ladle/react";
import ProfileIcon from ".";
import { SPOOKY_GHOST_IMAGE_BASE64 } from "@/.ladle/constants";

interface StoryProps {
  badgeValue: number;
  trimColor: string;
}

function StoryComponent({ badgeValue, trimColor }: StoryProps) {
  return (
    <div className="bg-polarNight-1 p-5">
      <ProfileIcon
        icon={SPOOKY_GHOST_IMAGE_BASE64}
        badgeValue={badgeValue}
        trimColor={trimColor}
      />
    </div>
  );
}

export const Default: Story<StoryProps> = StoryComponent;

Default.args = {
  badgeValue: 5,
  trimColor: "#bf616a",
};

Default.argTypes = {
  badgeValue: {
    control: { type: "number" },
  },
  trimColor: {
    control: { type: "radio" },
    options: ["#bf616a", "#d08770", "#ebcb8b", "#a3be8c", "#b48ead"],
  },
};

export default {
  title: "Molecules / Profile Icon",
} satisfies StoryDefault;
