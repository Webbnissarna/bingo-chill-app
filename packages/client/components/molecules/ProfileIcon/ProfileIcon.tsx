import Base64Icon from "@/components/atoms/Base64Icon/Base64Icon";
import Text from "@/components/atoms/Text";

interface ProfileIconProps {
  icon: string;
  badgeValue: number;
  trimColor: string;
}

export default function ProfileIcon({
  icon,
  badgeValue,
  trimColor,
}: ProfileIconProps): JSX.Element {
  return (
    <div className="w-14 h-14 rounded-xl bg-polarNight-0 flex items-center justify-center relative">
      <div className="max-h-full aspect-square">
        <Base64Icon src={icon} description="profile" />
      </div>
      <div
        className="w-10/12 h-1 rounded-lg absolute bottom-0"
        style={{ backgroundColor: trimColor }}
      />
      <div className="w-8 h-8 bg-polarNight-3 rounded-full absolute -top-4 -right-4 flex items-center justify-center">
        <Text>{badgeValue}</Text>
      </div>
    </div>
  );
}
