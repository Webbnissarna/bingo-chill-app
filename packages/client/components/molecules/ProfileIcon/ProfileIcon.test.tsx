import { render } from "../../../.jest/render";
import ProfileIcon from ".";

describe("ProfileIcon", () => {
  it("renders", () => {
    const { asFragment } = render(
      <ProfileIcon name="" badgeValue={0} icon="" trimColor="" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
