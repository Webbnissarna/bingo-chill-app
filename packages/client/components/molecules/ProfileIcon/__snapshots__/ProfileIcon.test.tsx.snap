// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileIcon renders 1`] = `
<DocumentFragment>
  <div
    class="w-14 h-14 rounded-xl bg-polarNight-0 flex items-center justify-center relative"
  >
    <div
      class="max-h-full aspect-square"
    >
      <img
        alt="profile"
        src=""
      />
    </div>
    <div
      class="w-10/12 h-1 rounded-lg absolute bottom-0"
    />
    <div
      class="w-8 h-8 bg-polarNight-3 rounded-full absolute -top-4 -right-4 flex items-center justify-center"
    >
      <span
        class="font-text text-snowStorm-2 text-base"
      >
        0
      </span>
    </div>
  </div>
</DocumentFragment>
`;
